### ISSUE TITLE
Show more details if MinimumReplicasUnavailable

### ISSUE THREAD

**Issue by guettli** [guettli/check-conditions#11]
**Current state:** 💤 waiting for agent

[Open in AgentLoop UI](https://agentloop.thomas-guettler.de/issues/check-conditions/11)

## Summary

When a Deployment reports `Available=False` / `MinimumReplicasUnavailable`, the tool prints the generic message `"Deployment does not have minimum availability."` but not the actual replica counts. The user wants desired vs. available replicas shown. The numbers live on the same object we already have in hand (`obj.status`), so this is a display-only enrichment — no extra API calls.

Relevant code: `printConditions` in `pkg/checkconditions/checkconditions.go:604` builds each output line (`outLine`, line 674) and already has `obj unstructured.Unstructured` and `gvr` in scope.

<details>
<summary>📋 Planning questions (from the plan stage)</summary>

## Questions

1. Which replica numbers should the line include? A Deployment's `status` exposes `replicas`, `readyReplicas`, `availableReplicas`, `updatedReplicas`, `unavailableReplicas`, and desired lives in `spec.replicas`.
   Option A: just desired vs. available — e.g. `available 1/3`
   Option B: available + ready + updated vs. desired — e.g. `available 1/3 ready 1 updated 2`
   Option C: something else — tell me exactly which fields

2. How should the extra detail be rendered relative to the existing line format `... MinimumReplicasUnavailable "Deployment does not have minimum availability." (3m1s)`?
   Option A: append after the duration — `... "..." (3m1s) [available 1/3]`
   Option B: insert before the duration — `... "..." available 1/3 (3m1s)`
   Option C: something else

3. Should this enrichment apply only to Deployments, or also to other replica-backed workloads that carry `Available`/`MinimumReplicasUnavailable` (e.g. ReplicaSets, StatefulSets, DaemonSets)?
   Option A: Deployments only (matches the issue verbatim)
   Option B: all workload types that expose desired/available replica counts
   Option C: something else

4. Should the counts only be appended for the `MinimumReplicasUnavailable` reason, or for any unhealthy `Available=False` condition on the resource?
   Option A: only when reason is `MinimumReplicasUnavailable`
   Option B: any `Available=False` condition on a deployment
   Option C: whenever the resource is a deployment and has replica status, regardless of condition

</details>

## Plan (under assumptions: Option A for each question above)

1. **Add a replica-detail helper** in `pkg/checkconditions/checkconditions.go`:
   - `func deploymentReplicaDetail(obj unstructured.Unstructured) string` that reads `spec.replicas` (desired, default `1` when absent) and `status.availableReplicas` (default `0`) via `unstructured.NestedInt64`.
   - Return a formatted fragment like `available 1/3` (empty string if the numbers can't be read, so the line degrades gracefully).

2. **Wire it into `printConditions`** (around the `outLine` construction, line 674):
   - After building the merged row, when `gvr.Resource == "deployments"` and the reason is `MinimumReplicasUnavailable`, compute the detail and append it to `outLine` per the chosen format (Option A: after the duration).
   - Apply the same append to the per-type `singleLine` used for `--while` regex matching (line 683) so regex behavior stays consistent with what's printed. Keep the appended text stable/greppable.

3. **Guard against the merge path**: because rows sharing `(status, reason, message)` are merged and multiple deployments never merge into one line (they differ by object), the append stays per-object — no change to merge logic needed. Just ensure the detail is derived from `obj`, which is per-object.

4. **Tests** in `pkg/checkconditions/checkconditions_test.go`:
   - Add `TestPrintConditionsAppendsDeploymentReplicaDetail`: build an unstructured Deployment with `spec.replicas=3`, `status.availableReplicas=1`, and an `Available=False` / `MinimumReplicasUnavailable` condition; assert the emitted line contains `available 1/3`.
   - Add a negative case: a deployment with a different reason (or a non-deployment resource) produces no replica suffix.
   - Add a missing-fields case: `status.availableReplicas` absent → treated as `0` (`available 0/3`) and no panic.

5. **Docs**: no README change strictly required; the output format is self-explanatory. Update the README example only if the maintainer wants the new format documented (optional).

### Notes / edge cases
- `spec.replicas` absent means Kubernetes defaults to 1 — reflect that default.
- Numbers come from cached list output, so they may lag reality slightly; that's acceptable and consistent with how the rest of the tool reports status.

>   topolvm-system deployments topolvm-controller Condition Available=False MinimumReplicasUnavailable "Deployment does not have minimum availability." (3m1s)

I would like to see how many are desired, and how many are actual available.

<details>
<summary>⏱ Timeline (21 steps)</summary>

| When | Event | Duration | Waiting on |
|------|-------|----------|------------|
| … | +1 earlier event | | |
| 08-27 20:28 | label `loop/triage-in-process` set | 7m47s | 💤 waiting for agent |
| 08-27 20:35 | agent run started | 5m2s | ❌ agent failed |
| 08-27 20:41 | agent run failed | 1s | 💤 waiting for agent |
| 08-27 20:41 | agent run started | 1m6s | ▶️ agent working |
| 08-27 20:42 | label `NeedSupervisor` set | 4m19s | 💤 waiting for human |
| 08-27 20:46 | label `NeedSupervisor` cleared | 26h21m | 💤 waiting for agent |
| 08-28 23:08 | label `loop/triage` set | 17s | 💤 waiting for agent |
| 08-28 23:08 | label `loop/triage-in-process` set | 1s | 💤 waiting for agent |
| 08-28 23:08 | agent run started | 3m34s | ▶️ agent working |
| 08-28 23:11 | label `loop/plan` set | 0s | 💤 waiting for agent |
| 08-28 23:11 | agent run succeeded | 13s | 💤 waiting for agent |
| 08-28 23:12 | label `loop/plan-in-process` set | 1s | 💤 waiting for agent |
| 08-28 23:12 | agent run started | 6m4s | ✅ agent ran |
| 08-28 23:18 | agent run succeeded | 1s | 💤 waiting for agent |
| 08-28 23:18 | label `loop/plan-questions` set | 56s | 💤 waiting for agent |
| 08-28 23:19 | label `loop/code` set | 6s | 💤 waiting for agent |
| 08-28 23:19 | label `loop/code-in-process` set | 1s | 💤 waiting for agent |
| 08-28 23:19 | agent run started | 8m20s | ✅ agent ran |
| 08-28 23:27 | agent run succeeded | 0s | 💤 waiting for agent |
| 08-28 23:27 | label `loop/code-ci-pending` set | ongoing | 💤 waiting for agent |

</details>

**Comment by guettlibot** [comment]
agentloop: previous agent run did not complete; resuming acpx session (restart 1/3).

Reason:
agentloop: the agent run did not complete (no result was produced).

- [old run](https://agentloop.thomas-guettler.de/issues/check-conditions/11/runs/run-20260827T202812.435741349Z)
- [new run](https://agentloop.thomas-guettler.de/issues/check-conditions/11/runs/run-20260827T203058.010001167Z)

**Comment by guettlibot** [comment]
agentloop: previous agent run did not complete; resuming acpx session (restart 2/3).

Reason:
agentloop: Worker container "worker" terminated with exit code 1: run-issue: write preflight failure: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 10.43.159.30:8090: connect: connection refused"
{"time":"2026-08-27T20:35:08.324463543Z","level":"ERROR","msg":"agentloop failed","err":"run-issue: restore credential: write .claude/projects/-data-agentloop-agentloop-issues-1117/24901e7f-8bec-459f-908a-9c4e1875a803.jsonl: rpc error: code = Unavailable desc = error reading from server: connection reset by peer"}

- [old run](https://agentloop.thomas-guettler.de/issues/check-conditions/11/runs/run-20260827T203058.010001167Z)
- [new run](https://agentloop.thomas-guettler.de/issues/check-conditions/11/runs/run-20260827T203558.052986204Z)

**Comment by guettlibot** [comment]
agentloop: previous agent run did not complete; resuming acpx session (restart 3/3).

Reason:
agentloop: could not read the agent output: open /data/agentloop/check-conditions/issues/11.meta/run-20260827T203558.052986204Z/output.json: no such file or directory

- [old run](https://agentloop.thomas-guettler.de/issues/check-conditions/11/runs/run-20260827T203558.052986204Z)
- [new run](https://agentloop.thomas-guettler.de/issues/check-conditions/11/runs/run-20260827T204100.797265757Z)

**Comment by guettlibot** [comment]
This is a well-specified, actionable feature request. Let me record my triage decision.

The issue asks: when a Deployment reports `Available=False MinimumReplicasUnavailable`, augment the output with the desired and actual-available replica counts.

**Verification against `main`:**
- The condition output is built generically in `printConditions` at `pkg/checkconditions/checkconditions.go:674`, which only formats the condition type/status/reason/message — there is no logic anywhere that reads replica counts. A search for `availableReplicas`/`desired` (`checkconditions.go`) turns up nothing, so the requested detail is genuinely not present today.
- The underlying data (`spec.replicas`, `status.replicas`, `status.availableReplicas`) is already fetched — each object is a full `unstructured.Unstructured` in `printResources` (`checkconditions.go:538`), so the values are readily accessible via `unstructured.NestedInt64`.
- No open issue duplicates this. The closest replica-related entries (#879–881) are Longhorn ignore-regexes, unrelated to Deployment replica reporting.

The request is concrete (add desired/available replica counts to the line for Deployments with this condition), the data is already in hand, and the code has a clear insertion point — enough for a coding agent to plan.

<sub>agentloop · agent: claude</sub>

**Comment by guettlibot** [comment]
Answers — confirming Option A on all four (the scope that matches the issue and stays clean):

- **Q1 = A** — `available X/Y` (status.availableReplicas over spec.replicas). That single ratio directly explains *why* MinimumReplicasUnavailable fired; degrade to empty string if the numbers can't be read.
- **Q2 = A** — append after the duration, bracketed and greppable: `... "Deployment does not have minimum availability." (3m1s) [available 1/3]`. Apply the same append to the `singleLine` used for `--while` regex matching so printed and matched text stay consistent.
- **Q3 = A** — Deployments only. `MinimumReplicasUnavailable` is a Deployment-specific condition reason, so this is the correct and complete scope; other workloads use different reasons and would be a separate issue.
- **Q4 = A** — only when the reason is `MinimumReplicasUnavailable`.

Proceed with the plan as written (A/A/A/A).

**Comment by guettlibot** [comment]
Done. PR opened: https://github.com/guettli/check-conditions/pull/40 (Closes #11).

## What I did

When a Deployment reports `Available=False` / `MinimumReplicasUnavailable`, the output line now shows the actual replica counts:

```
  topolvm-system deployments topolvm-controller Condition Available=False MinimumReplicasUnavailable "Deployment does not have minimum availability." (3m1s) available 1/3
```

- Added `deploymentReplicaDetail(obj)` which reads `status.availableReplicas` (default `0`) and `spec.replicas` (default `1`, matching Kubernetes' default) from the object already in hand — no extra API calls.
- Wired it into `printConditions`: the `available <n>/<desired>` fragment is appended only for `deployments` with reason `MinimumReplicasUnavailable`, and degrades gracefully (empty string) if the numbers can't be read. It's applied to both the printed line and the per-type `--while` regex line so matching stays consistent.

## How I verified

Installed Go 1.24 locally and ran:
- `go test ./...` — passes, including three new tests (detail appended → `available 1/3`; missing `availableReplicas` → `available 0/3` with no panic; no suffix for other reasons).
- `go vet ./...` — clean.

CI (`go test ./...` + golangci-lint) will run on the PR.

<sub>agentloop · agent: claude</sub>

**Comment by guettlibot** [comment]
agentloop: closing this issue — its pull request (https://github.com/guettli/check-conditions/pull/40) has been merged.
